### PR TITLE
[WIP] Total, rain and ice reflectivity are added to the P3 std output

### PR DESCRIPTION
**Pull request by JS-WRF-SBM**

### CODE REVIEW

File: components/eam/src/physics/cam/micro_p3.F90 [hunk 1]
```diff
@@ -991,6 +991,11 @@ subroutine p3_main_part3(kts, kte, kbot, ktop, kdir, &
    real(rtype)    :: qm_incld     !in-cloud qm
    real(rtype)    :: bm_incld     !in-cloud bm
 
+   real(rtype), dimension(kts:kte) :: ze_rain, ze_ice
+
+   ze_rain = 1.e-22_rtype
```
Comment (PeterCaldwell) [inline components/eam/src/physics/cam/micro_p3.F90:996]: Shouldn't these initial values be 10*log10(1e-22*1e18)=40 for consistency with the min value that's actually imposed when calculations are done?

**Comment by PeterCaldwell** [comment]
This PR requires changes on the C++ side since it changes the number of outputs from `p3_main`. Note that it changes the variable names but not the actual arguments to `p3_main_part3` so no changes are needed for that proc. 

The harder way to implement needed changes would be to also have C++ also compute/pass these new output variables. The easier way is just to have the F90 unit test code read but discard these new outputs. This latter approach would mean that v1 couldn't output these new variables... but I think that's fine for the time being.

**Review by JS-WRF-SBM — COMMENTED** [review]
I hope you can change vars name in case needed.

**Comment by wrliugit** [comment]
Hi, I tried this PR in order to have some radar reflectivity results and found a minor bug. In components/eam/src/physics/cam/micro_p3.F90, lines 699-701, there should be a space. Sepcifically ,this means changing the line from 
call add_default('diag_equiv_reflectivity', 1, '')
to
call add_default('diag_equiv_reflectivity', 1, ' '). 
The code works after this change. Thanks.

**Comment by AaronDonahue** [comment]
@PeterCaldwell , is this still a bug we need to fix?

**Comment by PeterCaldwell** [comment]
I think this PR adds radar reflectivity as a microphysics output in v0 but we never integrated it because we don't have the same variable for v1. It is a good addition but we never had the time to prioritize adding it to v1. I realize now that we should have just made these outputs available but not output by default (by deleting the "add_default" lines). In that case, cprnc tests will pass and everything should be good. It would also be nice to have these outputs in v1, but again... not the most pressing to-do item.

**Comment by PeterCaldwell** [comment]
Ok, I've modified this PR so all it does is add v0 output variables diag_equiv_reflectivity, diag_ze_rain, and diag_ze_ice *when the user explicitly asks for them*. This should solve the problem that C++/F90 cprnc comparisons failed because of new variables in F90 that we didn't add to v1. Before integrating, we should sanity check that CIME v0 tests still pass and that a v0 run asking for these 3 outputs succeeds (which it should because Weiran has been running with this change for a while). I haven't run these tests yet. I'd also like a review from @wrliugit or @hassanbeydoun to make sure the reflectivity formulae make sense in p3_micro_part3. I'm pretty sure the piping in the rest of the PR is correct.

**Review by hassanbeydoun — APPROVED** [review]
Looks good!

**Comment by AaronDonahue** [comment]
@PeterCaldwell , can we turn this into a git-issue and fold it into the I/O epic about fields we want to add as possible outputs?

**Comment by PeterCaldwell** [comment]
Unless @hassanbeydoun cares, I think we should just close this issue. Apparently nobody cares about it enough to complain about it not being done for years.

**Comment by AaronDonahue** [comment]
@PeterCaldwell sounds good.  I'm going to close this for now and someone can reopen.

**Pull request closed without merging**